### PR TITLE
FW-3468 MP3 files and audio not playable inside chat widget after sending.

### PR DESCRIPTION
**Pull request by Prochnost55**
### What do you want to achieve?
- remove the bug where the encrypted audio was not playing in specific condition

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- open the chat widget
- create a new conversation
- send an audio as first message
- the audio was not playable previously

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
- The decryption logic was present inside the scroll eventListner. This eventListner was not triggering when the messages were less 

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -15113,6 +15093,38 @@ var userOverride = {
...
+                                KommunicateUI.processLazyImage(img, img.getAttribute('data-thumbnailBlobKey'))
+                            }
+                            break;
+                        case 'AUDIO':
+                            var video = encryptedElements[i];
+                            if (KommunicateUI.isInView(video, document.querySelector('#mck-message-cell .mck-message-inner'))) {
+                                KommunicateUI.processEncMedia(video, video.getAttribute('data-blobkey'))
+                            }
+                            break;
+                        case 'VIDEO':
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:15111]: whats the difference in Audio and video? it looks to be sae

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 2]
```diff
@@ -15113,6 +15093,38 @@ var userOverride = {
                     $file_progress.removeClass('n-vis').addClass('vis');
                 }
             };
+            _this.handleEncryptedElements = function(encryptedElements){
+                for(var i = 0; i < encryptedElements.length; i++){
+                    switch (encryptedElements[i].tagName) {
+                        case 'IMG':
+                            var img = encryptedElements[i];
+                            if (KommunicateUI.isInView(img, document.querySelector('#mck-message-cell .mck-message-inner'))) {
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:15101]: isInView code looks to be same in all conditions, if that's the case, then it should be outside of switch statement. and inside code should have only what is unique to that part

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 3]
```diff
@@ -4839,6 +4839,21 @@ var userOverride = {
                         _this.sendPriceMessage();
                     }
                 });
+                const MSG_LIST_MUTATION_OBSERVER = new MutationObserver(function(mutationsList, observer){
+                    // callback to be executed when a childList mutation happens.
+                    var encryptedElements = document.querySelectorAll('.file-enc');
+                    if(!encryptedElements.length){
+                        return;
+                    };
+                    for(const mutation of mutationsList) {
```
Comment (gargshantnu) [inline webplugin/js/app/mck-sidebox-1.0.js:4848]: var

Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js]: updated

**Review by gargshantnu — APPROVED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 1 above)
Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js:15106]: no there was no difference.
Previously I didn't knew how to use two values for one switch case statement
Have update the code now.
Thanks

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 2 above)
Comment (Prochnost55) [reply webplugin/js/app/mck-sidebox-1.0.js]: moved it out

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 4]
```diff
@@ -15113,6 +15093,32 @@ var userOverride = {
                     $file_progress.removeClass('n-vis').addClass('vis');
                 }
             };
+            _this.handleEncryptedElements = function(encryptedElements){
+                for(var i = 0; i < encryptedElements.length; i++){
+                    var isElementInView = KommunicateUI.isInView(encryptedElements[i], document.querySelector('#mck-message-cell .mck-message-inner'))
+                    switch (encryptedElements[i].tagName) {
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:15099]: is executing switch required if if (isElementInView) is false?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by gargshantnu — APPROVED** [review]

**Pull request merged**